internal/controller/storageclient_controller.go

```diff
@@ -421,9 +423,51 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	//Once the owner references are removed from resources, StorageClaim is deleted
+	if err = r.deleteStorageClaims(); err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
Is there a reason the clean up is at the end of the reconcile phases? Doesn't it mean that we will not clean up unless we managed to go through an entire successful reconcile? Wouldn't it make more sense to have this cleanup at the start of the reconcile?